### PR TITLE
Update readme and add example

### PR DESCRIPTION
**Pull request by eclipse1985**
This PR adds a more complete README and one full example

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,88 +1,167 @@
...
+        console.error(message);
+    }
+}
+
+ArduinoIoTCloud.connect(options).then(() => {
+    console.log("Connected to Arduino IoT Cloud broker");
+}, error => {
+    console.error(error);
 });
+```
```
Comment (pirropirro) [inline README.md:62]:
```suggestion
ArduinoIoTCloud.connect(options)
  .then(() => console.log("Connected to Arduino IoT Cloud broker"))
  .catch(error => console.error(error));
```

File: README.md [hunk 2]
```diff
@@ -1,88 +1,167 @@
...
+- The ID of the *Thing* the *property* belongs to. You can list all your things and properties using the [Javascript Rest API client](https://www.npmjs.com/package/@arduino/arduino-iot-client), calling the [GET Things endpoint](https://www.arduino.cc/reference/en/iot/api/index.html#api-ThingsV2-thingsV2List)
+- The *variable name* of the property you want to listen
 
-// Register a callback on a property value change
-// 
-ArduinoCloud.onPropertyValue(thingId, propertyName, updateCb).then(() => {
-  // updateCb(message) will be called every time a new value is available. Value can be string, number, or a boolean depending on the property type
+```javascript
+const { ArduinoIoTCloud } = require('arduino-iot-js');
+const thingID = "THING_ID"
```
Comment (pirropirro) [inline README.md:72]: typo
```suggestion
const thingId = "THING_ID"
```

File: README.md [hunk 3]
```diff
@@ -1,88 +1,167 @@
...
+    console.log("Connected to Arduino IoT Cloud broker");
+    ArduinoIoTCloud.onPropertyValue(thingId, variableName, showUpdates = value => {
+        console.log(value);
+    }).then(() => {
+        console.log("Callback registered");
+    }, error => {
+        console.log(error);
+    });
 });
+```
```
Comment (pirropirro) [inline README.md:93]:
```suggestion
ArduinoIoTCloud.connect(options)
  .then(() => {
    console.log("Connected to Arduino IoT Cloud broker");
    return ArduinoIoTCloud.onPropertyValue(thingId, variableName,
      value => console.log(value));
  })
  .then(() => console.log("Callback registered"))
  .catch(error => console.log(error));
```

File: README.md [hunk 4]
```diff
@@ -1,88 +1,167 @@
...
 
-// Re-connect with a new authentication token, keeping the subscriptions
-// to the Things topics
-ArduinoCloud.updateToken(newToken).then(() => {
-  // Successful reconnection with the provided new token
+### How to disconnect from Arduino IoT Cloud Broker
+```javascript
+ArduinoCloud.disconnect().then(() => {
+  console.log("Successfully disconnected");
 });
```
Comment (pirropirro) [inline README.md:100]:
```suggestion
ArduinoCloud.disconnect()
  .then(() => console.log("Successfully disconnected"));
```

File: README.md [hunk 5]
```diff
@@ -1,88 +1,167 @@
...
-
 ```
+### How to send property values to the device
+To do this you need:
+- The ID of the *Thing* the *property* belongs to. You can list all your things and properties using the [Javascript Rest API client](https://www.npmjs.com/package/@arduino/arduino-iot-client),  calling the [GET Things endpoint](https://www.arduino.cc/reference/en/iot/api/index.html#api-ThingsV2-thingsV2List)
+- The *variable name* of the property you want to set
+- Value can be either a string, a boolean or a number
+```javascript
+const { ArduinoIoTCloud } = require('arduino-iot-js');
+const thingID = "THING_ID"
```
Comment (pirropirro) [inline README.md:109]: typo
```suggestion
const thingId = "THING_ID"
```

File: README.md [hunk 6]
```diff
@@ -1,88 +1,167 @@
...
+        console.error(message);
+    }
+}
+
+ArduinoIoTCloud.connect(options).then(() => {
+    console.log("Connected to Arduino IoT Cloud broker");
+    ArduinoCloud.sendProperty(thingId, variableName, value).then(() => {
+        console.log("Property value correctly sent");
+    });    
+});
```
Comment (pirropirro) [inline README.md:125]:
```suggestion
ArduinoIoTCloud.connect(options)
  .then(() => {
    console.log("Connected to Arduino IoT Cloud broker");
    return ArduinoCloud.sendProperty(thingId, variableName, value).
      then(() => console.log("Property value correctly sent"));
  })
  .catch(error => console.log(error));
```

File: README.md [hunk 7]
```diff
@@ -1,88 +1,167 @@
...
+                    console.log(property.variable_name+": "+value)
+                }).then(() => {
+                    console.log("Callback registered for "+property.variable_name);
+                });
+            });
+        });
+    });
+}, error  => {
+    console.log(error)
+});
```
Comment (pirropirro) [inline README.md:166]:
```suggestion
ArduinoIoTCloud.connect(options)
  .then(() => {
    console.log("Connected to Arduino IoT Cloud MQTT broker");

    // Init Arduino API Client
    const ArduinoIoTClient = ArduinoIoTApi.ApiClient.instance;
    ArduinoIoTClient.authentications['oauth2'].accessToken = ArduinoIoTCloud.getToken();

    const thingsApi = new ArduinoIoTAPI.ThingsV2Api(ArduinoIoTClient);
    const propertiesAPI = new ArduinoIoTApi.PropertiesV2Api(ArduinoIoTClient);

    return thingsApi.thingsV2List()
      .then(things => {
        things.forEach(thing => {
          propertiesAPI.propertiesV2List(thing.id)
            .then(properties => {
              properties.forEach(property => {
                ArduinoIoTCloud.onPropertyValue(thing.id, property.variable_name,
                  value => console.log(property.variable_name + ": " + value))
                  .then(() => console.log("Callback registered for " + property.variable_name))
                  .catch(error => console.log(error));
              });
            })
            .catch(error => console.log(error));
        });
      });
  })
  .catch(error => console.log(error));
```

File: README.md (hunk 3 above)
Comment (eclipse1985) [reply README.md:90]: I think that using an anonymous function as property update callback doesn't work

**Pull request merged**